torch/csrc/autograd/function.h

```diff
@@ -555,6 +562,34 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
     return false;
   }
 
+#ifdef TORCH_COMPILED_AUTOGRAD
+  // see [Note: Compiled Autograd]
+  // Used by compiled autograd to
+  //   1) Extract tensors/symint args
+  //   2) Collect node information for specialization and caching
```

Comment:
Could you please document in more details how CompiledNodeArgs should be used here and how (in particular what needs to be collected as you explained in another comment on the PR)?